src/nrnoc/init.cpp

```diff
@@ -772,30 +768,24 @@ namespace {
  *
  * This logic used to live inside hoc_register_dparam_semantics.
  */
+
+// name to int map for the negative types
+// xx_ion and #xx_ion will get values of type and type+1000 respectively
+static std::unordered_map<std::string, int> name_to_negint = {{"area", -1},
```

Comment:
Wonder if this is the best way to associate semantic name to the negative semantic enumeration value.

Reply:
I guess good enough for now. Perhaps can refactor so defined in only one place later and magic numbers never used. Reopen if disagree.

Reply:
do we need those semantic names as strings? if not, I would rather go with an `enum class`:

```c++
enum class name_to_negint  {
    area = -1,
    iontype = -2,
    cvodeieq = -3,
    netsend = -4,
    pointer = -5,
    pntproc = -6,
    bbcorepointer = -7,
    watch = -8,
    diam = -9,
    fornetcon  = -10,
    random = -11
};
```
and define that in a header where this can be used everywhere (over time).

Reply:
did you decide to defer this to later?